crates/fuel-core/src/graphql_api/storage/da_compression/reverse_key.rs

```diff
@@ -0,0 +1,82 @@
+use fuel_core_types::{
+    fuel_tx::{
+        input::PredicateCode,
+        ScriptCode,
+    },
+    fuel_types::{
+        Address,
+        AssetId,
+        Bytes32,
+        ContractId,
+    },
+};
+use std::ops::Deref;
+
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Eq,
+    serde::Serialize,
+    serde::Deserialize,
+    strum::EnumCount,
+)]
+/// The reverse key for the temporal registry index.
+/// By this key we can find the registry key from the temporal registry.
+pub enum ReverseKey {
+    Address(Address),
+    AssetId(AssetId),
+    ContractId(ContractId),
+    /// Hash of the script code.
+    ScriptCode(Bytes32),
+    /// Hash of the predicate code.
+    PredicateCode(Bytes32),
+}
+
+impl From<&Address> for ReverseKey {
+    fn from(address: &Address) -> Self {
+        Self::Address(*address)
+    }
+}
+
+impl From<&AssetId> for ReverseKey {
+    fn from(asset_id: &AssetId) -> Self {
+        Self::AssetId(*asset_id)
+    }
+}
+
+impl From<&ContractId> for ReverseKey {
+    fn from(contract_id: &ContractId) -> Self {
+        Self::ContractId(*contract_id)
+    }
+}
+
+impl From<&ScriptCode> for ReverseKey {
+    fn from(script_code: &ScriptCode) -> Self {
+        let hash = fuel_core_types::fuel_crypto::Hasher::hash(script_code.deref());
```

Comment:
Why are we using on `fuel-crypto` for this hash? The hash here is implementation detail of the database, and has nothing to do with hashes used in the vm or the protocol.

Reply:
You are right, it is up to the database to decide how better to hash it=) 

What do you think about using blake2b instead of sha2?